src/disposable/AsyncDisposableSet.ts

```diff
@@ -0,0 +1,32 @@
+import { IAsyncDisposable } from './types';
+
+export class AsyncDisposableSet implements IAsyncDisposable {
+  protected readonly disposables: Set<IAsyncDisposable> = new Set();
```

Comment:
```suggestion
  protected readonly disposables = new Set<IAsyncDisposable>();
```
less code, but can stay as is